src/one/nio/net/Session.java

```diff
@@ -232,7 +232,7 @@ public void handleException(Throwable e) {
         } else if (e instanceof SSLException) {
             if (log.isDebugEnabled()) log.debug("SSL/TLS failure: {}", getRemoteHost());
         } else {
-            log.error("Cannot process session from {}", getRemoteHost(), e);
+            if (log.isErrorEnabled()) log.error("Cannot process session from {}", getRemoteHost(), e);
```

Comment:
Usually error logging is enabled, so this check seems unnecessary.

Reply:
Here `getRemoteHost()` call is non-trivial and would happen always thus it has to be explicitly guarded.

Reply:
I get your point and I would agree with you in case of `info()`, but `error()` is supposed to be always enabled. Usages of `isErrorEnabled()` doesn't happen often in practice.

Reply:
Done